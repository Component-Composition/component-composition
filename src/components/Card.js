import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ _id, name, image, gifLink }) => {

  return (
    <div key={_id}>
      {name && <header><h3>{name}</h3></header>}
      <figure>
        <img src={image || gifLink} alt={name} />
      </figure>
      <p>{name} has {_id}</p>
    </div>
  );
};

Card.propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string,
  image: PropTypes.string,
  gifLink: PropTypes.string
};

export default Card;
