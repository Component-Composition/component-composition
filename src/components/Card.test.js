import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card Component', () => {
  it('renders Card', () => {
    const wrapper = shallow(<Card _id={'5da237699734fdcb7bef8f51'} name={'Arnold Shortman'} image={'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
