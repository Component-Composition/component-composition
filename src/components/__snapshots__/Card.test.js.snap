// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders Card 1`] = `
<div
  key="5da237699734fdcb7bef8f51"
>
  <header>
    <h3>
      Arnold Shortman
    </h3>
  </header>
  <figure>
    <img
      alt="Arnold Shortman"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
    />
  </figure>
  <p>
    Arnold Shortman
     has 
    5da237699734fdcb7bef8f51
  </p>
</div>
`;
